### PR TITLE
Rework User Interface

### PR DESCRIPTION
**Pull request by schweikart**
This PR introduces a uniform layout to all pages and re-orders some of the elements to make the application more intuitive.

### CODE REVIEW

File: src/components/landing-page/ProblemCard.tsx [hunk 1]
```diff
@@ -4,8 +4,7 @@ import NextLink from "next/link";
 interface ProblemCardProps {
   href: string;
   new: boolean;
-  qubits: number;
-  speedup: "polynomial" | "superpolynomial";
+  tags: string[];
```
Comment (Elscrux) [inline src/components/landing-page/ProblemCard.tsx:7]: We should maybe add documentation for the props. Not necessary to merge this PR, just in general

File: src/components/solvers/TextInputMask.tsx [hunk 2]
```diff
@@ -3,13 +3,10 @@ import Head from "next/head";
 import React, { ReactElement, useState } from "react";
 import { Container } from "../Container";
 import { Main } from "../Main";
-import { InputButtonPanel } from "./buttons/InputButtonPanel";
 import { Help } from "./SAT/Help";
-import { SolverTitle } from "./SolverTitle";
+import { EditorControls } from "./EditorControls";
 
 export interface TextInputMaskProperties {
```
Comment (Elscrux) [inline src/components/solvers/TextInputMask.tsx:9]: We should maybe add documentation for all props in this interface. Not necessary to merge this PR, just in general

File: src/pages/solve/FeatureModelAnomaly.tsx [hunk 3]
```diff
@@ -22,31 +33,49 @@ const FeatureModelAnomaly: NextPage = () => {
...
-              onChange={setSelectedAnomalies}
-              labelledBy="Select anomalies"
-            />
-          </Box>
+    <Layout>
+      <Heading as="h1">FeatureModel Anomaly Solvers</Heading>
+      <Text color="text" align="justify">
+        For a tree of features with cross tree constraints, these solvers can
+        detect various anomalies. Feature Models are given in the Universal
+        Variablity language.
```
Comment (Elscrux) [inline src/pages/solve/FeatureModelAnomaly.tsx]: Imo we should add the abbreviation, UVL, here. People might only be familiar with that.

Comment (Elscrux) [reply src/pages/solve/FeatureModelAnomaly.tsx]: We also support Extended UVL btw

File: src/pages/solve/FeatureModelAnomaly.tsx [hunk 4]
```diff
@@ -22,31 +33,49 @@ const FeatureModelAnomaly: NextPage = () => {
...
-          <Box width="300px">
-            <MultiSelect
-              options={anomalies}
-              value={selectedAnomalies}
-              onChange={setSelectedAnomalies}
-              labelledBy="Select anomalies"
-            />
-          </Box>
+    <Layout>
+      <Heading as="h1">FeatureModel Anomaly Solvers</Heading>
```
Comment (Elscrux) [inline src/pages/solve/FeatureModelAnomaly.tsx]: Any reason why FeatureModel has no space in between now?

File: src/pages/solve/MaxCut.tsx [hunk 5]
```diff
@@ -18,29 +27,45 @@ const MaxCut: NextPage = () => {
...
-            <GraphArea
-              graphData={graphData}
-              graphHeight={500}
-              graphWidth={500}
-            />
-          </Center>
+    <Layout>
+      <Heading as="h1">MaxCut Solver</Heading>
+      <Text color="text" align="justify">
+        For a given undirected, weighted graph, this algorithm finds a partition
```
Comment (Elscrux) [inline src/pages/solve/MaxCut.tsx]: We also support graphs that are not weighted, maybe add that too

File: src/pages/solve/MaxCut.tsx [hunk 6]
```diff
@@ -18,29 +27,45 @@ const MaxCut: NextPage = () => {
...
+        of the graph so that the sum of edge weights <i>between</i> these
+        partitions is maximal. The graph can be specified in the{" "}
+        <Link href="https://en.wikipedia.org/wiki/Graph_Modelling_Language">
+          Graph Modeling Language
+        </Link>{" "}
+        and edge weights should be specified with the <Code>weight</Code>{" "}
```
Comment (Elscrux) [inline src/pages/solve/MaxCut.tsx]:
```suggestion
        and edge weights can be specified with the <Code>weight</Code>{" "}
```

File: src/pages/solve/MaxCut.tsx (hunk 5 above)
Comment (Elscrux) [inline src/pages/solve/MaxCut.tsx]:
```suggestion
        For a given (un)directed, (weighted) graph, this algorithm finds a partition
```
We could almost cut directed and weighted at that point as we support both variants each.

File: src/pages/solve/SAT.tsx [hunk 7]
```diff
@@ -1,80 +1,65 @@
...
       </Head>
-      <SolverTitle
-        title="SAT Solver"
-        text="For a given Boolean formula, this algorithm checks if there exists an interpretation that satisfies it."
+
+      <Heading as="h1">SAT Solver</Heading>
+      <Text color="text" align="justify">
+        For a given Boolean formula, this algorithm checks if there is an
+        interpretation that satisfies it. You can enter any boolean formula with
+        any number of variables and combine them with boolean operators.
```
Comment (Elscrux) [inline src/pages/solve/SAT.tsx]:
```suggestion
        any number of variables and combine them with boolean operators (and, or, not).
```

File: src/pages/solve/SAT.tsx [hunk 8]
```diff
@@ -1,80 +1,65 @@
...
+      </Text>
+
+      <Spacer />
+
+      <EditorControls
+        idleText={'Try "a and not (not a or not b)" 👇'}
```
Comment (Elscrux) [inline src/pages/solve/SAT.tsx]:
```suggestion
        idleText={'Try "a and (not a or not b)" 👇'}
```

Comment (Elscrux) [reply src/pages/solve/SAT.tsx]: Currently the website has a problem this with formula

File: src/pages/index.tsx [hunk 9]
```diff
@@ -1,119 +1,58 @@
...
+      </Text>
+      <Text color="text" align="justify">
+        Our GitHub:{" "}
+        <Link href="https://github.com/ProvideQ" color="blue.400">
+          @ProvideQ
+        </Link>
+        <br />
+        API documentation:{" "}
+        <Link
+          href={process.env.NEXT_PUBLIC_API_BASE_URL + "/swagger-ui/index.html"}
```
Comment (Elscrux) [inline src/pages/index.tsx]: This could also use baseUrl()

File: src/components/landing-page/ProblemCard.tsx (hunk 1 above)
Comment (schweikart) [reply src/components/landing-page/ProblemCard.tsx:7]: now tracked in https://github.com/ProvideQ/toolbox-web/issues/32

File: src/components/solvers/TextInputMask.tsx (hunk 2 above)
Comment (schweikart) [reply src/components/solvers/TextInputMask.tsx:9]: now tracked in https://github.com/ProvideQ/toolbox-web/issues/32

File: src/pages/solve/MaxCut.tsx (hunk 5 above)
Comment (schweikart) [reply src/pages/solve/MaxCut.tsx]: I'll remove the `undirected` adjective completely

File: src/pages/solve/SAT.tsx (hunk 8 above)
Comment (schweikart) [reply src/pages/solve/SAT.tsx]: btw, this is tracked in https://github.com/ProvideQ/toolbox-web/issues/29

**Review by Elscrux — APPROVED** [review]

**Pull request merged**